ui/src/TestAzureTreeDataProvider.ts

```diff
@@ -0,0 +1,233 @@
+/*---------------------------------------------------------------------------------------------
+ *  Copyright (c) Microsoft Corporation. All rights reserved.
+ *  Licensed under the MIT License. See License.txt in the project root for license information.
+ *--------------------------------------------------------------------------------------------*/
+
+import { SubscriptionClient } from 'azure-arm-resource';
+import { SubscriptionListResult, TenantListResult } from 'azure-arm-resource/lib/subscription/models';
+import { ApplicationTokenCredentials, AzureEnvironment, loginWithServicePrincipalSecret } from 'ms-rest-azure';
+import { Disposable, Event, EventEmitter, TreeItem, TreeItemCollapsibleState } from 'vscode';
+import TelemetryReporter from 'vscode-extension-telemetry';
+import { IActionContext, IAzureNode, IAzureParentTreeItem, IAzureTreeDataProvider, IAzureUserInput, IChildProvider } from '../index';
+import { AzureSession } from './azure-account.api';
+import { callWithTelemetryAndErrorHandling } from './callWithTelemetryAndErrorHandling';
+import { ArgumentError } from './errors';
+import { localize } from './localize';
+import { parseError } from './parseError';
+import { AzureNode } from './treeDataProvider/AzureNode';
+import { AzureParentNode } from './treeDataProvider/AzureParentNode';
+import { LoadMoreTreeItem } from './treeDataProvider/LoadMoreTreeItem';
+import { RootNode } from './treeDataProvider/RootNode';
+import { SubscriptionNode } from './treeDataProvider/SubscriptionNode';
+
+export class TestAzureTreeDataProvider implements IAzureTreeDataProvider<IAzureNode> {
+    public static readonly subscriptionContextValue: string = SubscriptionNode.contextValue;
+
+    private _onDidChangeTreeDataEmitter: EventEmitter<IAzureNode> = new EventEmitter<IAzureNode>();
+    private _onNodeCreateEmitter: EventEmitter<IAzureNode> = new EventEmitter<IAzureNode>();
+
+    private readonly _loadMoreCommandId: string;
+    private _resourceProvider: IChildProvider;
+    private _ui: IAzureUserInput;
+    private _customRootNodes: AzureNode[];
+    private _telemetryReporter: TelemetryReporter | undefined;
+
+    private _subscriptionNodes: IAzureNode[] = [];
+    private _disposables: Disposable[] = [];
+
+    constructor(resourceProvider: IChildProvider, loadMoreCommandId: string, ui: IAzureUserInput, telemetryReporter: TelemetryReporter | undefined, rootTreeItems?: IAzureParentTreeItem[]) {
+        this._resourceProvider = resourceProvider;
+        this._loadMoreCommandId = loadMoreCommandId;
+        this._ui = ui;
```

Comment:
What is IAzureUserInput used for here?  Not sure I like the reassignInputs() method.

Reply:
Today there is no way to add more inputs to a TestUserInput once it is created.  I wanted to implement this as a way to add/alter fake user inputs for tests if I need to dynamically add inputs.  The only other alternative to that today is to create a new TestUserInput object.